src/bin/kawari-world.rs

```diff
@@ -68,20 +74,368 @@ fn spawn_main_loop() -> (ServerHandle, JoinHandle<()>) {
+                                                    player_data = database.find_player_data(actor_id, &mut game_data);
+                                                }
+                                                connection.player_data = player_data;
+                                            }
+
+                                            // collect actor data
+                                            connection.initialize(actor_id).await;
+
+                                        }
+                                        _ => panic!("initial_setup: The zone connection type must start with a Setup segment! What happened? Received: {segment:#?}")
+                                    }
+                                }
+                                spawn_client(connection);
+                                break;
+                            } else if header.connection_type == ConnectionType::Chat {
+                                let state = ConnectionState::Zone {
+                                    clientbound_oodle: OodleNetwork::new(),
+                                    serverbound_oodle: OodleNetwork::new(),
+                                    scrambler_keys: None,
+                                };
+
+                                let mut connection = ChatConnection {
+                                    config: get_config().world,
+                                    ip,
+                                    id,
+                                    actor_id: 0,
+                                    state,
+                                    last_keep_alive: Instant::now(),
+                                    socket,
+                                };
+
+                                // Handle setup before passing off control to the chat connection.
+                                let (segments, _) = connection.parse_packet(&buf[..n]);
+                                for segment in segments {
+                                    match &segment.data {
+                                        SegmentData::Setup { actor_id } => {
+                                            // for some reason they send a string representation
+                                            let actor_id = actor_id.parse::<u32>().unwrap();
+                                            connection.actor_id = actor_id;
+                                            connection.initialize().await;
+                                        }
+                                        _ => panic!("initial_setup: The chat connection type must start with a Setup segment! What happened? Received: {segment:#?}")
+                                    }
+                                }
+                                spawn_chat_connection(connection);
+                                break;
+                            } else {
+                                panic!("Connection type is None! How did this happen?");
```

Comment:
Let's switch this to a tracing::warn!, because this could let someone miscraft a packet that takes down the server. (Or a misbehaving client xD)

Reply:
Done!